code/game/objects/items/weapons/weapon.dm

```diff
@@ -14,6 +14,7 @@
 	var/max_shots = 0 //codex, energy weapons
 	var/scope_zoom = FALSE//codex
 	var/self_recharge = FALSE //codex
+	var/green_blood_drained = 20
```

Comment:
Missing autodoc + if you are changing this due to attack speed why is the resource calculation not just updated to incorporate attackspeed

Reply:
Sure